backend/src/main/java/com/redhat/cloud/notifications/routers/internal/errata/ErrataMigrationRepository.java

```diff
@@ -0,0 +1,104 @@
+package com.redhat.cloud.notifications.routers.internal.errata;
+
+import com.redhat.cloud.notifications.models.EventType;
+import com.redhat.cloud.notifications.models.EventTypeEmailSubscription;
+import com.redhat.cloud.notifications.models.EventTypeEmailSubscriptionId;
+import com.redhat.cloud.notifications.models.SubscriptionType;
+import io.quarkus.logging.Log;
+import jakarta.enterprise.context.ApplicationScoped;
+import jakarta.inject.Inject;
+import jakarta.persistence.EntityManager;
+import jakarta.transaction.Transactional;
+import jakarta.validation.ConstraintViolationException;
+
+import java.util.List;
+
+@ApplicationScoped
+public class ErrataMigrationRepository {
+    /**
+     * The maximum size of the batch of subscriptions we want to insert in the
+     * database.
+     */
+    private static final int INSERTION_BATCH_SIZE = 500;
+
+    /**
+     * The internal name of the Errata application in our database.
+     */
+    public static final String ERRATA_APPLICATION_NAME = "errata-notifications";
+
+    @Inject
+    EntityManager entityManager;
+
+    /**
+     * Finds all the Errata event types in the database.
+     * @return the list of Errata event types in the database.
+     */
+    protected List<EventType> findErrataEventTypes() {
+        final String query =
+            "FROM " +
+                "EventType AS et " +
+            "INNER JOIN " +
+                "et.application AS applications " +
+            "WHERE " +
+                "applications.name = :applicationName";
+
+        return this.entityManager
+            .createQuery(query, EventType.class)
+            .setParameter("applicationName", ERRATA_APPLICATION_NAME)
+            .getResultList();
+    }
+
+    /**
+     * Saves the given Errata subscriptions in the database.
+     * @param errataSubscriptions the list of the Errata subscriptions to save
+     *                            in the database.
+     */
+    @Transactional
+    protected void saveErrataSubscriptions(final List<ErrataSubscription> errataSubscriptions) {
+        Log.infof("Errata subscriptions' migration begins with a batch size of %s", INSERTION_BATCH_SIZE);
```

Comment:
Should we empty the table first in case we need to run the migration more than once?

Reply:
Hmmm... If we clean up the table wouldn't we delete the email subscriptions that existed before the migration? I was thinking that maybe we can do an "ON CONFLICT DO NOTHING" insert so that we can safely run it multiple times. Or manually check for that in the code and log a "duplicated email subscription found. Doing nothing" kind of thing too.

Reply:
Emptying the table is indeed not an option. Why did I even suggest that? :sweat_smile:

Anyway, being able to run the migration more than once would be great. I'm fine with ignoring records that are already in the DB.

Reply:
Perfect, changed that!